pkg/sql/parser/datum.go

```diff
@@ -199,6 +200,18 @@ func ParseDBool(s string) (*DBool, error) {
 	return MakeDBool(DBool(b)), nil
 }
 
+// ParseDByte parses a string representation of hex encoded binary data.
+func ParseDByte(s string) (*DBytes, error) {
+	if len(s) > 2 && (s[0] == '\\' && (s[1] == 'x' || s[1] == 'X')) {
+		hexstr, err := hex.DecodeString(s[2:])
+		if err != nil {
+			return NewDBytes(DBytes(s)), nil
```

Comment:
Sorry I was slow to look at this -- this does introduce a minor incompatibility with Postgres, which will return an error if you send it invalid hex - e.g. `e'\\xthisisnthex'`. I'm not sure how much this matters, but I feel like we might as well fix it anyway.